projects/Open-Rival-test/src/tests/TestWalkerComponent.cpp

```diff
@@ -32,10 +33,63 @@ SCENARIO("WalkerComponent can plan a route", "[components][movement-component]")
             {
                 REQUIRE(!walkerComponent->getRoute().isEmpty());
                 MapNode expectedDestination = { 1, 2 };
-                REQUIRE(walkerComponent->getRoute().getDestination() == expectedDestination);
+                REQUIRE(walkerComponent->getRoute().getFinalDestination() == expectedDestination);
+            }
+        }
+    }
+    
+    GIVEN("A unit with a WalkerComponent in an enclosed space")
+    {
+        Entity e(EntityType::Unit, 1, 1);
+        e.attach(std::make_shared<WalkerComponent>());
+        e.onSpawn(&world, 0, { 3, 2 });
+
+        WalkerComponent* walkerComponent = e.requireComponent<WalkerComponent>(WalkerComponent::key);
+
+        MapNode enclosure[] = {
+            { 1, 2 }, { 2, 2 }, { 3, 1 }, { 4, 1 }, { 2, 3 }, { 5, 1 }, { 3, 3 }, { 4, 3 }, { 5, 2 }, { 6, 2 } 
+        };
```

Comment:
It's hard to make sense of this without drawing it out.

This is fine for now, but we may want to come up with a better solution in the future. Unfortunately, the zigzagging nature of the tile grid makes it impractical to create a map from a string, but maybe we can define a list of offsets like this:

```
const MapNode[] offsets = {
    { Facing::North },
    { Facing::NorthEast },
    { Facing::East, Facing::East }, // travel East twice
    // etc.
};
```